### ISSUE TITLE
Update and clarify Elastic Agent proxy docs

### ISSUE THREAD

**Issue by AndersonQ** [elastic/ingest-docs#198]
The Elastic Agent takes up to 3 different proxies:
 - for fleet-server,
 - for the output and
 - a default one used for the all internet connections, more specifically to download artifacts from `artifacts.elastic.co`.

How to configure the proxies changes if the agent is fleet-managed or stand-alone. Also with the introduction of the beta Proxy feature on Fleet>Settings, it's unclear for the customer that what is shown in the docs still valid or not.

Finally the information hierarchy isn't quite clear, for example _"Set the proxy for communicating with Elasticsearch"_ seems to be under _"Set Elastic Agent proxy settings in a standalone agent policy"_. Which should not be the case. Also on _"Set the proxy for communicating with Elasticsearch"_ shows how to do so for both, the stand-alone and fleet-managed agent.

In summary this request is for improve the [Use a proxy server with Elastic Agent and Fleet](https://www.elastic.co/guide/en/fleet/8.7/fleet-agent-proxy-support.html#fleet-agent-proxy-support) page by:
 - reviewing the information hierarchy
 - split more clearly what applies to stand alone agent and fleet managed
 - include the new beta Proxy feature on Fleet>Settings

**Comment by TheRiffRafi** [comment]
It doesn't seem that this actually applies to artifacts:

>a default one used for the all internet connections, more specifically to download artifacts from artifacts.elastic.co.

See [here](https://github.com/elastic/ingest-docs/pull/96#issuecomment-1569145955)

**Comment by AndersonQ** [comment]
> It doesn't seem that this actually applies to artifacts:
> 
> > a default one used for the all internet connections, more specifically to download artifacts from artifacts.elastic.co.
> 
> See [here](https://github.com/elastic/ingest-docs/pull/96#issuecomment-1569145955)

It does, your comment is correct. It isn't possible to define a proxy just to download artifacts. What is possible is to set a global proxy and override this proxy for fleet-server and ES.

**Comment by jstrassb** [comment]
Thanks for filing this issue, indeed it would be much appreciated! 🙇‍♂️ 

A question that came up is if we only support uppercase environment variables here, or also the lower case alternatives?
I see only upper case mentioned and documented, though it would help to clearly state if that is the case.

It can easily lead to confusion and @lucabelluccini pointed me to  https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ for a related discussion, e.g. what exactly can we use, how does it behave, etc. would be great to add!

**Comment by AndersonQ** [comment]
I believe the best is to stick to the upper case version of them. As it's right now, both would work for the agent at least, however I'm not sure if the other process managed by the agent all support both versions.

**Comment by vishaangelova** [comment]
Hey @AndersonQ, could you have a look at this issue? It was opened quite some time ago, and we’ve since moved to the new v3 docs. The original asciidoc was split up based on our current information architecture, so currently we have these docs:

[Using a proxy server with Elastic Agent and Fleet](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-support)
├ [When to configure proxy settings](https://www.elastic.co/docs/reference/fleet/elastic-agent-proxy-config)
├ [Proxy server connectivity using default host variables](https://www.elastic.co/docs/reference/fleet/host-proxy-env-vars)
├ [Fleet-managed Elastic Agent connectivity using a proxy server](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-managed)
├ [Standalone Elastic Agent connectivity using a proxy server](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-standalone)
├ [Set the proxy URL of the Elastic Package Registry](https://www.elastic.co/docs/reference/fleet/epr-proxy-setting)

> - reviewing the information hierarchy
> - split more clearly what applies to stand alone agent and fleet managed

Does our current structure resolve these points?

> - include the new beta Proxy feature on Fleet>Settings

The default proxy setup in Fleet is described in [Configuring proxy servers in Fleet for managed agents](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-support#fleet-agent-proxy-server-managed-agents), so I think this point should be covered now.

> - Clarify which components run by the agent are affected by the HTTP_PROXY/HTTPS_PROXY/NO_PROXY environment variables. (see https://github.com/elastic/elastic-agent/issues/2602)

Can you also check if this part of the request is still relevant or if it’s not already clear from the [related doc](https://www.elastic.co/docs/reference/fleet/host-proxy-env-vars)?

As for the environment variables, currently the doc only describes uppercase env. variables, so I’m not sure if there’s a need to add anything for that.

**Comment by AndersonQ** [comment]
I've been away from the agent team for a while, so I believe it's better to ask someone form the elastic agent control plane to check it as well. Anyway, here are my thoughts.

I think it mostly addressed the original intent of the issue. 
As I was checking, I mage some note, see below.

Anyway, as I said before, I think it's better to ask someone form the @elastic/elastic-agent-control-plane to check it.


> [Using a proxy server with Elastic Agent and Fleet](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-support)

In the note: there are support for mTLS, as far as I understand it's considered a form of authentication.

> [Proxy server connectivity using default host variables](https://www.elastic.co/docs/reference/fleet/host-proxy-env-vars)

The note at the end, it seems the proxy settings explained do not affect the connections to `artifacts.elastic.co`. If I'm not mistake, they do. But better double-check with the agent team.

> [Fleet-managed Elastic Agent connectivity using a proxy server](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-managed)

the 1st warning: It's correct, but, as far as I remember, the agent checks the new connection before applying it and only applies if it works. Reading it seemed to me the agent blindly applies any new config.

steps 5 and 6: I understood the docs would change the config for an existing policy. Then items 5 and 6 seems odd in this context. 

step 6: perhaps mention when enrolling the agent a the proxy can be passed through the CLI, then once it connects to fleet-server and downloads the policy, it'll use the proxy from the policy.

After point 6, there is _"[Set the proxy for retrieving agent policies from Fleet](https://www.elastic.co/docs/reference/fleet/fleet-agent-proxy-managed#cli-proxy-settings)"_:
Perhaps it can put more emphases it's about the CLI command to install the agent.

**Comment by vishaangelova** [comment]
Thanks! In that case, let me first ask for a review from the Elastic Agent team...

@ebeahan - this is low priority, I think, but when your team has some capacity, could we have someone from your team have a look at [this comment](https://github.com/elastic/ingest-docs/issues/198#issuecomment-3660493051) and at [Anderson’s notes](https://github.com/elastic/ingest-docs/issues/198#issuecomment-3670377526) and clarify if we need to make changes to the docs? Thank you!
